### PR TITLE
Filter BUILD_SOURCEVERSIONAUTHOR in sameenv dshell test

### PR DESCRIPTION
**Pull request by wilzbach**
See e.g. https://github.com/dlang/phobos/pull/7342#issuecomment-571154708

### CODE REVIEW

File: test/dshell/extra-files/printenv.d [hunk 1]
```diff
@@ -1,7 +1,15 @@
 import std.array, std.stdio, std.process, std.algorithm;
+
+auto filterEnv(T)(T env)
+{
+    // Windows shell output is not in UTF-8
+    // https://github.com/dlang/phobos/pull/7342#issuecomment-571154708
+    return env.filter!(e => e.key.among("BUILD_SOURCEVERSIONAUTHOR"));
```
Comment (Biotronic) [inline test/dshell/extra-files/printenv.d]: Maybe add a bang here, so it shows everything except `BUILD_SOURCEVERSIONAUTHOR`, rather than only that?

**Comment by WalterBright** [comment]
@wilzbach what's up with the buildkite failures?

**Comment by Biotronic** [comment]
Why the hell was this merged? As pointed out almost two months ago above, the filter removes the things it should keep, and keeps the things it should remove.

**Comment by MoonlightSentinel** [comment]
Fixup: #10919

**Pull request merged**